src/refda-ion/main.c

```diff
@@ -182,6 +182,16 @@ int main(int argc, char *argv[])
         {
             CACE_LOG_INFO("Agent startup completed");
         }
+
+        if (refda_agent_init_objs(&agent))
```

Comment:
If this occurs after `refda_agent_start()` it will need to be thread safe and lock the agent objects while manipulating them. It could also be refactored into agent start and occur before thread spawning. A longer-term behavior will be to only init if persisted object state #7 is not present.

Reply:
For now, added locking to `refda_agent_init_objs` to guarantee the necessary thread safety.